common/djangoapps/django_comment_common/models.py

```diff
@@ -105,3 +106,29 @@ class Meta:
 
     def __unicode__(self):
         return self.name
+
+
+def all_permissions_for_user_in_course(user, course_id):  # pylint: disable=invalid-name
+    """Returns all the permissions the user has in the given course."""
+    course = modulestore().get_course(course_id)
+    if course is None:
+        raise ItemNotFoundError(course_id)
+    course_blacked_out = not course.forum_posts_allowed
+
+    all_roles = {role.name for role in Role.objects.filter(users=user, course_id=course_id)}
+
+    def permission_blacked_out(p_name):
+        """Returns true if the given permission is blacked out"""
+        return (
+            course_blacked_out and
+            all_roles == {FORUM_ROLE_STUDENT} and
+            (p_name.startswith('edit') or p_name.startswith('update') or p_name.startswith('create'))
+        )
+
+    permissions = {
+        permission.name
+        for permission
+        in Permission.objects.filter(roles__users=user, roles__course_id=course_id)
+        if not permission_blacked_out(permission.name)
```

Comment:
Actually, @benmcmorran I just understood that the code you pointed to is not deleted by this PR. Can you do some refactoring so that this logic only exists in one place? It seems brittle enough as it is-- we don't want 2 copies of it.